### DISCUSSION TITLE
company directory - city field filter

### DISCUSSION THREAD

**Discussion by LohanT49** [microsoft-search/pnp-modern-search#2916] [Q&A] [answered]
Hello,

We are looking to make a business directory using pnp modern search.
We can't add a second part in the filters.
We currently have the services and we want to add the cities as well.
However we don't know how to get this field or if it is possible.

Thank you for your help.

**Comment by kasperbolarsen** [comment, marked as answer]
Hi @LohanT49 
Try combining https://www.techmikael.com/2014/11/how-to-make-sure-user-profile.html and https://www.rlvision.com/blog/how-to-create-searchable-user-profile-properties-in-sharepoint-online/   and perhaps use this script to reindex the user profiles : https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps

**Reply by kasperbolarsen** [reply]
You might also have a look at https://github.com/microsoft-search/pnp-modern-search/discussions/2910

**Comment by jussihi** [comment]
@LohanT49 , did you manage to do this? I'm facing the exact same requirement today.
